src/content/docs/en/guides/content-collections.mdx

```diff
@@ -249,6 +249,7 @@ const dogs = defineCollection({
 
 export const collections = { blog, dogs };
 ```
+If the property is not defined in the Zod schema, it will be excluded from the collection entry data.
```

Comment:
Maybe it could be better highlighted as a note or a caution, eg.

![image](https://github.com/user-attachments/assets/83fb5cd1-1880-492f-8374-fa5fbd612a14)

Reply:
Nice suggestion. Initially I got the same idea. I just kept it just normal text because I am not fully aware about writing style of astro docs when to use note or caustion or normal text. I will wait for maintainers feeback on this.

Reply:
Thanks @apatel369 !

Also, I've been recently thinking, in terms of functionality, if it is not rather about _including_ than _excluding_ when speaking of schema related data, would make sense, eg.

![image](https://github.com/user-attachments/assets/36758b7f-50e9-448d-9d07-c5e60034ffcf)

, if it is functionally correct that schema potentially filters out undeclared properties.

Reply:
Thanks Sarah, that's a good point. The mentioned line tbh actually beats me, because I have questions about the functionality itself, and also it might be a tough nut for my english the way I see it atm.

The only thing I can tell about that sentence is that it tells me I have to use `zod syntax` to define schemas, but not about the "including only" or "excluding" behavior on actual data based on the defined schema, which I don't know if it's a self given functionality of schemas in general, an astro feature or a potential bug.

Reply:
@jurajkapsz Sorry for the confusion. I hear you and appreciate your feedback. I've also read your comment on the original issue. I will investigate this further and get back to you soon with more information.

Reply:
> The only thing I can tell about that sentence is that it tells me I have to use zod syntax to define schemas,

Yes, this has been our recommendation as this is what guarantees the typesafety. I guess what you've both discovered here is that it's *possible* to have a property not defined by Zod, but then it technically does not become part of your collection data.

I'm wondering how documenting this is a benefit? Is there a helpful use case for knowing that untyped data can "exist but not be used"? Sometimes in docs we don't mention everything that is possible, but describe the "happy path" simple case for everyone. It seemed to me like the original note was intended to be a kind of warning that you need to define every property, or it won't be part of your collection. If the goal is to tell/remind people to make Zod definitions, then saying "Every property in your collection must have a Zod definition" is a pretty clear instruction to do that on its own, I think? If you read that, you're left concluding that you need to make a Zod definition and... that's what we want! 😄 In which case, I'm not sure any change to existing documentation is needed?

If the intention is to say, "Don't worry if you can't define everything! You can still *have* extra data that's not defined, it just won't be a part of your collection. This is useful when..." Then THAT is a different story, but I'd need to know how that is a popular/helpful use case. It seems that this is not the intended use of content collections, and we typically don't document "possible but unsupported/unrecommended use." But if this use is helpful to people who otherwise couldn't have collections, then I get saying something.

So I guess it all comes down to the original intention of this PR?

Reply:
Thanks @apatel369 , @sarah11918 ! I was writing back, but I think I've found what it's all about, citing:

> By default, Zod object schemas strip out unrecognized keys during parsing. This means that if a user provides data with extra or unknown properties, Zod will remove them, conforming to the defined schema.

See also [.passthrough](https://zod.dev/?id=passthrough).

So this is by the looks the behavior I ran into. I don't think it is particularly mentioned in [astro] docs (at least not content collections), but it's probably a common knowledge about Zod I wasn't aware of - haven't used Zod as a lib before.

I am not sure if this can or should be part of Astro docs, nor some runtime logs.

I'd like to appreciate very much for your help Arpan and Sarah!

Reply:
@jurajkapsz  Glad that you found it. Since it is a Zod thing, I don't think we need to add it in Astro docs.

Reply:
Just noting that I left a comment in the original Astro issue, as I suspect this is in fact better supported by an error message (if that's technically possible!) rather than a change to docs.

My full comment is here! https://github.com/withastro/astro/issues/12404#issuecomment-2473469287